plugins/inputs/kafka_consumer/kafka_consumer.go

```diff
@@ -225,6 +257,80 @@ func (k *KafkaConsumer) Stop() {
 	k.wg.Wait()
 }
 
+func (k *KafkaConsumer) addTopicRegexps() error {
+	// We instantiate a new generic Kafka client, so we can ask
+	// it for all the topics it knows about.  Then we build
+	// regexps from our strings, loop over those, loop over the
+	// topics, and if we find a match, add that topic to
+	// out topic set, which then we turn back into a list at the end.
+	var err error
+	var allTopics []string
+	var exists = struct{}{}
+	regexps := make([]regexp.Regexp, 0, len(k.TopicRegexps))
+
+	kafkaClient, err := sarama.NewClient(k.Brokers, k.config)
+	if err != nil {
+		return err
+	}
+	defer kafkaClient.Close()
+	allTopics, err = kafkaClient.Topics()
+	if err != nil {
+		return err
+	}
+	for _, r := range k.TopicRegexps {
+		re := regexp.MustCompile(r)
+		regexps = append(regexps, *re)
+	}
```

Comment:
Can we please do this in `Init()` instead of compiling the regexp over and over? Furthermore, `MustCompile` will panic if the regexp (coming from a user) is invalid... Please change this and handle the error gracefully.

Reply:
Moved to Init() and failure to compile generates a log error but doesn't stop us from working, we just don't match on that particular RE.